debezium-connector-postgres/src/main/java/io/debezium/connector/postgresql/RecordsSnapshotProducer.java

```diff
@@ -116,59 +117,60 @@ protected void stop() {
             cleanup();
         }
     }
-    
+
     private void cleanup() {
         currentRecord.set(null);
         executorService.shutdownNow();
     }
-    
+
     private void takeSnapshot(Consumer<SourceRecord> consumer) {
         long snapshotStart = clock().currentTimeInMillis();
         Connection jdbcConnection = null;
         try (PostgresConnection connection = taskContext.createConnection()) {
             jdbcConnection = connection.connection();
             String lineSeparator = System.lineSeparator();
-    
+
             logger.info("Step 0: disabling autocommit");
             connection.setAutoCommit(false);
-    
+
             long lockTimeoutMillis = taskContext.config().snapshotLockTimeoutMillis();
-            logger.info("Step 1: starting transaction and refreshing the DB schemas for database '{}' and user '{}'", 
+            logger.info("Step 1: starting transaction and refreshing the DB schemas for database '{}' and user '{}'",
                         connection.database(), connection.username());
             // we're using the same isolation level that pg_backup uses
             StringBuilder statements = new StringBuilder("SET TRANSACTION ISOLATION LEVEL SERIALIZABLE, READ ONLY, DEFERRABLE;");
             connection.executeWithoutCommitting(statements.toString());
             statements.delete(0, statements.length());
-    
-            //next refresh the schema which will load all the tables taking the filters into account 
+
+            //next refresh the schema which will load all the tables taking the filters into account
             PostgresSchema schema = schema();
             schema.refresh(connection, false);
-    
+
             logger.info("Step 2: locking each of the database tables, waiting a maximum of '{}' seconds for each lock",
                         lockTimeoutMillis / 1000d);
             statements.append("SET lock_timeout = ").append(lockTimeoutMillis).append(";").append(lineSeparator);
             // we're locking in SHARE UPDATE EXCLUSIVE MODE to avoid concurrent schema changes while we're taking the snapshot
             // this does not prevent writes to the table, but prevents changes to the table's schema....
+            // DBZ-298 Quoting name in case it has been quoted originally; it doesn't do harm if it hasn't been quoted
             schema.tables().forEach(tableId -> statements.append("LOCK TABLE ")
-                                                         .append(tableId.toString())
+                                                         .append(tableId.toQuotedString('"'))
```

Comment:
What do you think about having `toDoubleQuotedString()` that basically calls `toQuotedString('"')`, and `toSingleQuotedString()` that calls `toQuotedString('\'')`? I suspect that would be easier to read here.